src/WebJobs.Script.WebHost/Security/KeyManagement/BaseSecretsRepository.cs

```diff
@@ -78,6 +87,11 @@ protected void OnChanged(object sender, FileSystemEventArgs e)
 
         public abstract Task<string[]> GetSecretSnapshots(ScriptSecretsType type, string functionName);
 
+        protected virtual void LogErrorMessage(string operation)
```

Comment:
I do not think we need `LogErrorMessage` method at all in base class. Lets use `BlobStorageSecretSasRepoError` and `BlobStorageSecretRepoError` directly from secret managers.